### DISCUSSION TITLE
Running DAG Tasks are stuck in queued status - Celery Flower Task state is "FAILURE"

### DISCUSSION THREAD

**Discussion by mihaly-hazag** [apache/airflow#29889] [Q&A]
Running Airflow in production on Rancher in dedicated containers for:
- scheduler
- flower
- worker
- webserver

When I run `example_bash_operator` sample DAG, its tasks are stuck in `queued` status. There is no info in the task logs, and there are no error messages in the container logs.

Opening flower I can see that all of the tasks are failed with the following error message:

```Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/airflow/executors/celery_executor.py", line 52, in execute_command
    subprocess.check_call(command, shell=True)
  File "/usr/local/lib/python3.6/subprocess.py", line 311, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['airflow', 'tasks', 'run', 'example_bash_operator', 'also_run_this', 'manual__2023-03-03T03:24:01.955384+00:00', '--local', '--subdir', 'DAGS_FOLDER/airflow_bash_dag.py']' returned non-zero exit status 2.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/celery/app/trace.py", line 412, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/celery/app/trace.py", line 704, in __protected_call__
    return self.run(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/airflow/executors/celery_executor.py", line 55, in execute_command
    raise AirflowException('Celery command failed')
airflow.exceptions.AirflowException: Celery command failed
```

Additional info:
- image: apache/airflow:2.5.1-python3.10, there is no modification on the image, no additional packages.
- everything works well locally using the Airflow provided docker-cmpose.yaml ('https://airflow.apache.org/docs/apache-airflow/2.5.1/docker-compose.yaml'). I cannot use the same in the Rancher environment, it does not support this version.
- for every container I defined the AIRFLOW__CELERY__BROKER_URL in environment variable (RabbitMQ), but it does not work. Celery Flower throws error: `No hostname was supplied. Reverting to default 'localhost'`
- Celery Flower works only if I define hostname with BROKER_URL in environment variable 

Please help me what can be the problem, why do I have failed tasks in Flower.

**Comment by potiuk** [comment]
Hard to say - no experience with rancher. but you should likely take a look at whether your containers are run properly. Whether the entrypoints and variables are confgured properly. Your best approach is to enter the running container in rancher and try to reproduce the what you see  as the command `airlfow tasks run ...` using the very same environment you have in rancher - same entrypoint, same configuration. You can run it manually to check - just be aware of the difference between interactive and non-interactive shell (for example certain initialization of variables might not happen in non-interactive mode, depends on your .rc files.  Example (random - found by google) explanation - https://www.vanimpe.eu/2014/01/18/different-shell-types-interactive-non-interactive-login/ and you can find more about it.

It just needs debugging of your deployment, and I am afraid you are the only one that can do it.

**Comment by mihaly-hazag** [comment]
I have tried `airflow task run` before and it works on all of the 4 containers (scheduler, flower, webserver, worker)

```
airflow@81aa7015bb2f:/opt/airflow$ airflow tasks run example_bash_operator this_will_skip scheduled__2023-02-28T00:00:00+00:00 --local --subdir DAGS_FOLDER/airflow_bash_dag.py
/home/airflow/.local/lib/python3.10/site-packages/airflow/models/base.py:49 MovedIn20Warning: [0m[31mDeprecated API features detected! These feature(s) are not compatible with SQLAlchemy 2.0. [0m[32mTo prevent incompatible upgrades prior to updating applications, ensure requirements files are pinned to "sqlalchemy<2.0". [0m[36mSet environment variable SQLALCHEMY_WARN_20=1 to show all deprecation warnings.  Set environment variable SQLALCHEMY_SILENCE_UBER_WARNING=1 to silence this message.[0m[0m (Background on SQLAlchemy 2.0 at: https://sqlalche.me/e/b8d9)
[2023-03-03 18:33:38,828] {dagbag.py:538} INFO - Filling up the DagBag from /opt/airflow/dags/airflow_bash_dag.py
[2023-03-03 18:33:41,343] {task_command.py:388} INFO - Running <TaskInstance: example_bash_operator.this_will_skip scheduled__2023-02-28T00:00:00+00:00 [skipped]> on host 81aa7015bb2f

```
